src/main/java/it/gov/pagopa/pu/classification/service/assessments/AssessmentsServiceImpl.java

```diff
@@ -81,6 +81,9 @@ public List<Assessments> createAssessment(Long receiptId, String operatorExterna
       .map(i -> {
         DebtPositionTypeOrg debtPositionTypeOrg = debtPositionTypeOrgService.getDebtPositionTypeOrgByInstallmentId(i.getInstallmentId(), accessToken);
 
+        // delete all existing assessment details for the given installment (iuv, iud) and debtPositionTypeOrg code
+        assessmentsDetailService.deleteAssessmentDetail(debtPositionTypeOrg.getCode(), i.getIuv(), i.getIud());
```

Comment:
here is wrong! as stated on JIRA task, we should perform the delete on all installments! not just on installment that now has a balance! you should a further map before the filter